pkg/version/plugin3/commonSpecForContainer.go

```diff
@@ -0,0 +1,112 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package plugin3
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/vmware/vic/lib/migration/manager"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/vmware/vic/lib/migration/errors"
+	"github.com/vmware/vic/pkg/trace"
+	"github.com/vmware/vic/pkg/vsphere/extraconfig"
+	"github.com/vmware/vic/pkg/vsphere/session"
+)
+
+const (
+	version = 3
+	target  = manager.ContainerConfigure
+)
+
+func init() {
+	defer trace.End(trace.Begin(fmt.Sprintf("Registering plugin %s:%d", target, version)))
+	if err := manager.Migrator.Register(version, target, &AddCommonSpecForContainer{}); err != nil {
+		log.Errorf("Failed to register plugin %s:%d, %s", target, version, err)
+	}
+}
+
+// AddCommonSpecForVM is plugin for vic 0.8.0-GA version upgrade
+type AddCommonSpecForContainer struct {
+}
+
+type ExecutorConfig struct {
+	Common `vic:"0.1" scope:"read-only" key:"common"`
+}
+
+type Common struct {
+	// A reference to the components hosting execution environment, if any
+	ExecutionEnvironment string
+
+	// Unambiguous ID with meaning in the context of its hosting execution environment
+	ID string `vic:"0.1" scope:"read-only" key:"id"`
+
+	// Convenience field to record a human readable name
+	Name string `vic:"0.1" scope:"read-only" key:"name"`
+
+	// Freeform notes related to the entity
+	Notes string `vic:"0.1" scope:"hidden" key:"notes"`
+}
+
+type NewExecutorConfig struct {
```

Comment:
@emlin with this code as written, would we wind up with objects actually named `NewExecutorConfig` elsewhere in the code?